### PR TITLE
fix transformblock calls, remove tf function

### PR DESCRIPTION
**Pull request by zmjjmz**
- `TransformBlock` was being called incorrectly
  - `norm_type` wasn't being provided, but the positional args call meant that it was using the `batch_momentum`. This effectively prevented batch_norm from being used
  - Once this was fixed, I noticed that `groups` and `virtual_batch_size` were being mixed up, so I switched the argument order (in the function definition due to laziness)
- While this should probably just use tfa directly, there was a [small change](https://github.com/tensorflow/addons/commit/e55167c7e16dc8945809bcc96dd335e72ebf0fa1#diff-f05bc2d568594ae2ac58eaf41b38752aL85) there to remove the `tf.function` decorator so that's replicated here.

Both of these were done in the hopes it would help https://github.com/titu1994/tf-TabNet/issues/9, but I haven't observed any significant change there (even using the new batch normalization)

### CODE REVIEW

**Comment by titu1994** [comment]
Thanks for looking into this ! Once again, I find that positional args are a curse. Thanks for correcting this defect. 

There's no need to remove the function decorators - they only compile the function. It does not affect mathematical calculation itself. Once that is reverted, I can merge this.

**Comment by zmjjmz** [comment]
Sure, was just trying to keep it in line with the updates to the tfa version.

**Comment by titu1994** [comment]
Yup no worries, and thank you again for your help !
